core_lib/src/structure/layer.cpp

```diff
@@ -213,72 +213,46 @@ bool Layer::removeKeyFrame(int position)
     return true;
 }
 
-bool Layer::moveKeyFrameForward(int position)
+bool Layer::moveKeyFrame(int position, int offset)
 {
-    return swapKeyFrames(position, position + 1);
-}
+    bool moved = false;
+    if (position == 1 && offset < 0) { return false; }
 
-bool Layer::moveKeyFrameBackward(int position)
-{
-    if (position != 1)
-    {
-        return swapKeyFrames(position, position - 1);
+    if (swapKeyFrames(position, position + offset)) {
+        moved = true;
+    } else {
+        setFrameSelected(position, true);
+        if (moveSelectedFrames(offset)) {
+            moved = true;
+        }
+        setFrameSelected(position + offset, false);
```

Comment:
This approach changes the behavior of the move keyframe shortcut and causes a lot of unwanted behavior when there is already one or more frames selected. Try selecting one or multiple frames and then moving with the cursor on or off a selected frame, with and with out a frame at the destination. There's different behavior in nearly every case now. In v0.6.5 it strictly moves the current keyframe, and I think would should probably just stick with the behavior. I'm fine with changing it to involve the selected frames, but only if we do it in some consistent, sane manner.

Reply:
Right, that's true.. I like this approach because the movement behavior is consistent with moving frames on the timeline but I can see that there could be some problems with doing it like this... For now I think that saving the selection, deselecting all frames and then re-selecting them afterwards is good enough.